lib/memtar.rb

```diff
@@ -16,7 +16,7 @@ def add_file path, content, opts = {}
     size = content.bytesize
     @io.write header opts.merge size: size, name: path
     @io.write content
-    @io.write "\0" * (512 - size % 512)
+    @io.write "\0" * ((512 - size % 512) % 512)
```

Comment:
Wouldn't  `"\0" * (size % 512)` accomplish the same thing?

Reply:
Close, but not quite:

| Size | Current <br/> (`512 - size % 512`) | Change <br /> (`(512 - size % 512) % 512`) | Suggested <br /> (`size % 512`) |
| ---- | ---------------------------- | ----------------------------------- | ------------------------ |
| 0    | 512                          | 0                                   | 0                        |
| 100 | 412 | 412 | 100 |
| 256 | 256 | 256 | 256 |
| 400 | 112 | 112 | 400 |
| 512 | 512 | 0 | 0 |

Reply:
Ah my bad, read too quick.